### DISCUSSION TITLE
Is it possible to prevent getting manually added steps/activities?

### DISCUSSION THREAD

**Discussion by Duskernel** [doville/rn-fitness-tracker#136] [Q&A]
Hello,

I have started using this project some weeks back and love it so far for its simplicity. 
But one thing I noticed is that, out of the box when getting steps data I cannot see a clear way of preventing to fetch manually added activities? 

In both android and Ios it is possible to manually add any amount of steps in their respective apps, but in my use case I only want to fetch the actually walked steps for the day and display this data. It looks like it should be possible to use meta data or provide options when fetching the data in the original native apis or other react-native libraries to prevent this. 

Do we have something similar and I am just blind? 

Best regards

**Comment by lukebars** [comment]
Hi @redines!

Right now it's not available, yet we could add such feature. 

Just to clarify, you would like to filter out the data that was entered via Health app / Google fit or remove every app's input and leave it to the device completely?

**Reply by Duskernel** [reply]
Hello, 

Ah I see :) 
I would like to filter out unwanted data which a user can manually insert. 
It would be cool if it was possible to control this via an options parameter in the data fetching so that we always can choose when we want to filter out something specific or not

**Reply by lukebars** [reply]
So basically, you would like to receive raw data with sources and all the metadata? Other way, it would be quite hard to do this satisfy-all logic 🤔

**Reply by Duskernel** [reply]
not sure on the final solution and what can be done, but if I could do something similar to how health kit does it here:
https://github.com/agencyenterprise/react-native-health/blob/master/docs/getStepCount.md

it would be a great addition I believe

**Reply by lukebars** [reply]
Seems like it should be quite easy to implement, we'll take a look on Monday and if it's all good - will try to ship it next week 😊 still need to figure out the google fit case

**Reply by Duskernel** [reply]
ah cool! 
would be great if it is possible :) 
and yeah.. googles documentation is not very great...
